src/main/java/org/prebid/server/floors/PriceFloorFetcher.java

```diff
@@ -0,0 +1,320 @@
+        }
+        validatePriceFloorRules(priceFloorRules, fetchConfig);
+
+        return ResponseCacheInfo.of(priceFloorRules, cacheTtlFromResponse(httpClientResponse));
+    }
+
+    private void validatePriceFloorRules(PriceFloorRules priceFloorRules, AccountPriceFloorsFetchConfig fetchConfig) {
+        final PriceFloorData data = priceFloorRules.getData();
+
+        if (data == null) {
+            throw new PreBidException("Price floor rules data must be present");
+        }
+
+        if (CollectionUtils.isEmpty(data.getModelGroups())) {
+            throw new PreBidException("Price floor rules should contain at least one model group");
+        }
+
+        final Integer maxRules = resolveMaxRules(fetchConfig.getMaxRules());
+
+        CollectionUtils.emptyIfNull(data.getModelGroups()).stream()
+                .filter(Objects::nonNull)
+                .forEach(modelGroup -> validateModelGroup(modelGroup, maxRules));
+    }
+
+    private static int resolveMaxRules(Integer accountMaxRules) {
+        return Objects.equals(accountMaxRules, 0L) ? Integer.MAX_VALUE : accountMaxRules;
+    }
+
+    private static void validateModelGroup(PriceFloorModelGroup modelGroup, Integer maxRules) {
+        final Map<String, BigDecimal> values = modelGroup.getValues();
+        if (MapUtils.isEmpty(values)) {
+            throw new PreBidException(String.format("Price floor rules values can't be null or empty, but were %s",
+                    values));
+        }
+
+        if (values.size() > maxRules) {
+            throw new PreBidException(String.format("Price floor rules number %s exceeded its maximum number %s",
+                    values.size(), maxRules));
+        }
+    }
+
+    private Long cacheTtlFromResponse(HttpClientResponse httpClientResponse) {
+        final String cacheMaxAge = httpClientResponse.getHeaders().get(HttpHeaders.CACHE_CONTROL);
+        if (StringUtils.isNotBlank(cacheMaxAge) && cacheMaxAge.contains("max-age")) {
+            final String[] maxAgeRecord = cacheMaxAge.split("=");
+            if (maxAgeRecord.length == 2) {
+                try {
+                    return Long.parseLong(maxAgeRecord[1]);
+                } catch (NumberFormatException ex) {
+                    logger.warn(String.format("Can't parse Cache Control header '%s'", cacheMaxAge));
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private PriceFloorRules updateCache(ResponseCacheInfo cacheInfo,
+                                        AccountPriceFloorsFetchConfig fetchConfig,
+                                        String accountId) {
+
+        long maxAgeTimerId = createMaxAgeTimer(accountId, resolveCacheTtl(cacheInfo, fetchConfig));
+        final AccountFetchContext fetchContext =
+                AccountFetchContext.of(cacheInfo.getRules(), maxAgeTimerId);
+
+        if (cacheInfo.getRules() != null || fetchedData.get(accountId) == null) {
```

Comment:
Can we use `containsKey(..)` instead of `get(..)` ?

Reply:
Good catch, thx!